src/components/magic/adeptPowers/util/AdeptPowersUtil.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PowerDetailRow from '../PowerDetailRow';
+
+const bonusDown = (name, bonusToRemove, decreaseBy, decrementAugmented) => {
+	const trimmedName = name.match(/^([^(]*)(\s{0,1}\(.*\)){0,1}$/)[1];
+	const bonusAction = {
+		'Improved Physical Attribute': () => {
+			decrementAugmented({attribute: bonusToRemove, decreaseBy});
+		},
+		default: () => {
+			return null;
+		}
+	};
+
+	(bonusAction[trimmedName] || bonusAction.default)();
+};
+
+const bonusUp = (name, bonusToApply, incrementAugmented) => {
+	const bonusAction = {
+		'Improved Physical Attribute': () => {
+			incrementAugmented({attribute: bonusToApply});
+		},
+		default: () => {
+			return null;
+		}
+	};
+
+	return (bonusAction[name] || bonusAction.default)();
+};
+
+const generateAddActionDetails = (isMystic, power, pointsSpent, maxPoints, selectedOption, incrementAugmented) => {
+	if (pointsSpent + Number(power.points) <= maxPoints) {
+		const name = power.name + selectedOption;
+		const powerToAdd = Object.assign({}, power, {name});
+
+		if (powerToAdd.bonus) {
+			powerToAdd.bonus = selectedOption.replace(/[()]/g, '');
+			bonusUp(power.name, powerToAdd.bonus, incrementAugmented);
+		}
+
+		if (powerToAdd.levels === 'yes') {
+			powerToAdd.levels = 1;
+		} else {
+			powerToAdd.levels = 'N/A';
+		}
+
+		return {newSpell: powerToAdd, isMystic};
+	}
+	return '';
+};
+
+const generateRemoveActionDetails = (isMystic, power, index, decrementAugmented) => {
+	if (power.bonus !== 'N/A') {
+		bonusDown(power.name.match(/^([^(]*)(\s{0,1}\(.*\)){0,1}$/)[0], power.bonus, power.levels, decrementAugmented);
+	}
+	return {powerIndex: index, isMystic};
+};
+
+const raiseLevelAction = (isMystic, powerName, bonusOfPower, index, raisePower, incrementAugmented) => {
+	return () => {
+		bonusUp(powerName, bonusOfPower, incrementAugmented);
```

Comment:
I've learned that the correct way to fire an action that needs to have multiple effects in reducers is to actually have the redux action type listened for on multiple reducers.

Because I was the one that made this terrible design decision originally, I'll refactor it later. It'll probably have the bonus of simplifying the raiseLevelAction and lowerLevelAction later too.

Or, if you'd like to give it a shot... make a new action creator and give it a type like IncreaseAdeptAttribute or something, and have that affect the attributes and magic reducers.

Reply:
That makes sense.  I have time and got a second wind, so I'll take a stab.  Currently tests aren't running when I connect mapStateToProps, going to fix that, get my changes smoke tested and pushed out.

Reply:
Looking at this now, it seems the react example has you dispatching the two independently like it is now:
https://redux.js.org/docs/advanced/ExampleRedditAPI.html

For example switching subreddits:
handleChange(nextSubreddit) {
  this.props.dispatch(selectSubreddit(nextSubreddit))
  this.props.dispatch(fetchPostsIfNeeded(nextSubreddit))
}

Reply:
Sounds good.